tests/drivers/stepper/stepper_api/src/main.c

```diff
@@ -112,97 +138,244 @@ ZTEST_F(stepper, test_set_micro_step_interval_invalid_zero)
 
-	POLL_AND_CHECK_SIGNAL(
-		stepper_signal, stepper_event, STEPPER_EVENT_STEPS_COMPLETED,
-		K_MSEC(-steps * (100 + CONFIG_STEPPER_TEST_TIMING_TIMEOUT_TOLERANCE_PCT)));
+	POLL_AND_CHECK_SIGNAL(stepper_signal, stepper_event, STEPPER_EVENT_STEPS_COMPLETED,
+			      K_MSEC(STEPPER_TIMEOUT(-steps)));
+
 	(void)stepper_get_actual_position(fixture->dev, &steps);
-	zassert_equal(steps, -20u, "Target position should be %d but is %d", -20u, steps);
+	zassert_equal(steps, -10, "Target position should be %d but is %d", -10, steps);
+	zassert_equal(user_data_received, fixture->dev, "User data not received");
 }
 
-ZTEST_F(stepper, test_stop)
+ZTEST_F(stepper, test_move_by_is_moving_false_when_completed)
 {
+	int32_t steps = 20;
+	bool moving;
+
+	(void)stepper_set_microstep_interval(fixture->dev, CONFIG_STEPPER_TEST_MICROSTEP_INTERVAL);
 	(void)stepper_set_event_callback(fixture->dev, fixture->callback, (void *)fixture->dev);
+	(void)stepper_move_by(fixture->dev, steps);
 
-	/* Run the stepper in positive direction */
+	POLL_AND_CHECK_SIGNAL(stepper_signal, stepper_event, STEPPER_EVENT_STEPS_COMPLETED,
+			      K_MSEC(STEPPER_TIMEOUT(steps)));
+
+	k_msleep(1);
+	(void)stepper_is_moving(fixture->dev, &moving);
+	zassert_false(moving, "Driver should not be in state is_moving after completion");
+}
+
+ZTEST_F(stepper, test_move_by_zero_steps_no_movement)
+{
+	int32_t steps = 0;
+
+	(void)stepper_set_microstep_interval(fixture->dev, CONFIG_STEPPER_TEST_MICROSTEP_INTERVAL);
+	(void)stepper_set_event_callback(fixture->dev, fixture->callback, (void *)fixture->dev);
+	(void)stepper_move_by(fixture->dev, steps);
+
+	POLL_AND_CHECK_SIGNAL(stepper_signal, stepper_event, STEPPER_EVENT_STEPS_COMPLETED,
+			      K_NO_WAIT);
+
+	(void)stepper_get_actual_position(fixture->dev, &steps);
+	zassert_equal(steps, 0, "Target position should be %d but is %d", 0, steps);
+}
+
+ZTEST_F(stepper, test_move_by_is_moving_true_while_moving)
+{
+	int32_t steps = 20;
+	bool moving = false;
+
+	(void)stepper_set_microstep_interval(fixture->dev, CONFIG_STEPPER_TEST_MICROSTEP_INTERVAL);
+	(void)stepper_set_event_callback(fixture->dev, fixture->callback, (void *)fixture->dev);
+	(void)stepper_move_by(fixture->dev, steps);
+	(void)stepper_is_moving(fixture->dev, &moving);
+	zassert_true(moving, "Driver should be in state is_moving");
+}
+
+ZTEST_F(stepper, test_run_positive_direction_correct_position)
+{
+	int32_t steps = 0;
+	int32_t target_position = 5;
+
+	(void)stepper_set_microstep_interval(fixture->dev, CONFIG_STEPPER_TEST_MICROSTEP_INTERVAL);
 	(void)stepper_run(fixture->dev, STEPPER_DIRECTION_POSITIVE);
+	k_msleep(target_position * CONFIG_STEPPER_TEST_MICROSTEP_INTERVAL / NSEC_PER_MSEC);
 
-	/* Stop the stepper */
-	int ret = stepper_stop(fixture->dev);
-	bool is_moving;
+	(void)stepper_get_actual_position(fixture->dev, &steps);
+	int32_t min = target_position -
+		      ceil(((double)(CONFIG_STEPPER_TEST_TIMING_TIMEOUT_TOLERANCE_PCT * 5)) / 100);
```

Comment:
[DIV_ROUND_UP](https://docs.zephyrproject.org/apidoc/latest/group__sys-util.html#gae664e7492e37d324831caf2321ddda37) would work as well, or?

```suggestion
		      DIV_ROUND_UP(CONFIG_STEPPER_TEST_TIMING_TIMEOUT_TOLERANCE_PCT * 5,  100);
```